servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcFilters.java

```diff
@@ -59,19 +59,29 @@ public static StreamingHttpClientFilterFactory newGrpcDeadlineClientFilterFactor
      */
     public static StreamingHttpServiceFilterFactory newGrpcDeadlineServerFilterFactory(
             @Nullable Duration defaultTimeout) {
-        return new TimeoutHttpServiceFilter(grpcDetermineTimeout(
-                defaultTimeout == null ? null : ensurePositive(defaultTimeout, "defaultTimeout")), true);
+        return new TimeoutHttpServiceFilter(new GrpcDetermineTimeoutForRequestFunction(defaultTimeout), true);
     }
 
-    private static BiFunction<HttpRequestMetaData, TimeSource, Duration> grpcDetermineTimeout(
-            @Nullable Duration defaultTimeout) {
-        return (request, timeSource) -> {
-            /*
-             * Return the timeout duration extracted from the GRPC timeout HTTP header if present or default timeout.
-             *
-             * @param request The HTTP request to be used as source of the GRPC timeout header
-             * @return The non-negative timeout duration which may be null
-             */
+    private static final class GrpcDetermineTimeoutForRequestFunction
```

Comment:
Consider adding some documentation why this standalone class is there so noone by accident optimizes it back into a lambda.